### PR TITLE
upgrades: change drop of sys.jobs columns to hide

### LINKED ISSUE #121365: jobs: change 24.1 drop upgrade to rename+hide
The drop migration will rewrite the table which is unnecessarily expensive; all we need is to ensure they are not being accessed (by renaming it then hiding it)

NB: this blocks minting the cluster version.

Jira issue: CRDB-37215

### PR DESCRIPTION
**Pull request by dt**
The drop would have done a full rewrite of all the indexes of the table, building new indexes to swap in, which would be wasteful at best as these columns were already empty since 23.1 and any rows that had them would have expired 14d after completing. At worst, the full table rewrite would have had elevated risk of causing instability in the jobs system, and for little benefit as mentioned above.

This changes the drop to be rename+hide. The rename is required to ensure that any users using these defunct columnns mistakenly find out that they are defunct -- that was the intent of the DROP. The hide is cosmetic and makes interacting with the jobs table nicer / avoids needing to explain why the renamed columns are still there.

If/when we make a new jobs table we may want to drop these for real at the same time, but until then it just isn't worth the full rewrite.

Release note: none.
Epic: none.

### CODE REVIEW

**Review by stevendanna — APPROVED** [review]

File: pkg/upgrade/upgrades/v24_1_drop_payload_and_progress_jobs_test.go [hunk 1]
```diff
@@ -26,7 +26,7 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
-func TestDropPayloadAndProgressSystemJobs(t *testing.T) {
+func TestHidePayloadAndProgressSystemJobs(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (dt) [reply pkg/upgrade/upgrades/v24_1_drop_payload_and_progress_jobs_test.go:29]: Indeed, I was pleased

**Review by RaduBerinde — APPROVED** [review]
:lgtm:

> Reviewed 1 of 3 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/121604#-:-NuZKy5v2Li9NXBtjomM:bnfp4nl)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @dt, @kev-cao, @msbutler, and @stevendanna)

___
*[`pkg/upgrade/upgrades/v24_1_drop_payload_and_progress_jobs.go` line 21 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/121604#-NuZKMPDDmjkxGTmtD3E:-NuZKMPECK5IZwp9yXl0:bfw2a3s) ([raw file](https://github.com/cockroachdb/cockroach/blob/55608931a9591b502f3fb1de49a97ac13c3beb48/pkg/upgrade/upgrades/v24_1_drop_payload_and_progress_jobs.go#L21)):*
> ```Go
> 	"github.com/cockroachdb/cockroach/pkg/upgrade"
> )
> ```

[nit] should we also rename the file (`v24_1_hide_...`)
___
*[`pkg/upgrade/upgrades/v24_1_drop_payload_and_progress_jobs.go` line 33 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/121604#-NuZKeNi1GkZR1JvuvVX:-NuZKeNi1GkZR1JvuvVY:b-hpihbu) ([raw file](https://github.com/cockroachdb/cockroach/blob/55608931a9591b502f3fb1de49a97ac13c3beb48/pkg/upgrade/upgrades/v24_1_drop_payload_and_progress_jobs.go#L33)):*
> ```Go
> 		}
> 		for _, stmt := range []string{
> 			`ALTER TABLE system.jobs RENAME COLUMN payload TO dropped_payload`,
> ```

[nit] I'd rename to `payload_deprecated`. "dropped" is a misnomer here
___
*[`pkg/upgrade/upgrades/v24_1_drop_payload_and_progress_jobs_test.go` line 29 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/121604#gh-1549467208:-NuZKp7q8Ro8lgwJYi1l:b1tlmk4) ([raw file](https://github.com/cockroachdb/cockroach/blob/eff2dc68add0b212990a9bf67f459273270311bb/pkg/upgrade/upgrades/v24_1_drop_payload_and_progress_jobs_test.go#L29)):*
<details><summary><i>Previously, dt (David Taylor) wrote…</i></summary><blockquote>

Indeed, I was pleased
</blockquote></details>

We should verify that the columns are hidden.

**Review by fqazi — APPROVED** [review]
:lgtm:

> Reviewed 3 of 3 files at r1, 3 of 3 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/121604#-:-NuZLjLJ0YsiSGPIBBpV:bnfp4nl)* status: <span>:shipit:</span> complete! 2 of 0 LGTMs obtained (waiting on @dt, @kev-cao, @msbutler, and @stevendanna)

**Review by dt — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/121604)* status: <span>:shipit:</span> complete! 2 of 0 LGTMs obtained (waiting on @kev-cao, @msbutler, @RaduBerinde, and @stevendanna)

___
*[`pkg/upgrade/upgrades/v24_1_drop_payload_and_progress_jobs.go` line 33 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/121604#-NuZKeNi1GkZR1JvuvVX:-NuZUwAq0yMIUTFwc_Pk:b-fia2bd) ([raw file](https://github.com/cockroachdb/cockroach/blob/55608931a9591b502f3fb1de49a97ac13c3beb48/pkg/upgrade/upgrades/v24_1_drop_payload_and_progress_jobs.go#L33)):*
<details><summary><i>Previously, RaduBerinde wrote…</i></summary><blockquote>

[nit] I'd rename to `payload_deprecated`. "dropped" is a misnomer here
</blockquote></details>

Eh, most things we have that are labled "deprecated" are disfavored but still usable; this is not. I think "deprecated" is too weak a term -- we _want_ to consider it dropped: we don't read it, write it, or show it (unless we're showing hidden columns).

**Pull request merged**